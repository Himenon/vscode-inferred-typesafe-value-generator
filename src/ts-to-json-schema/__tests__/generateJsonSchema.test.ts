import * as path from "path";
import * as Module from "../index";

const basePath = __dirname;

describe("JSON Schema", () => {
  test("types.ts", () => {
    const schema = Module.generateJsonSchema({
      files: [path.join(basePath, "./types.ts")],
      typeName: {
        isUnknown: false,
        firstTypeName: "ServerSideConfig",
        isArray: false,
        intersectinoTypeNames: [],
        isIntersectionType: false,
      },
      basePath,
    });
    expect(schema).toMatchSnapshot();
  });

  test("sample.ts", () => {
    const schema = Module.generateJsonSchema({
      files: [path.join(basePath, "./sample.ts")],
      typeName: {
        isUnknown: false,
        firstTypeName: "ServerSideConfig",
        isArray: false,
        intersectinoTypeNames: [],
        isIntersectionType: false,
      },
      basePath,
    });
    expect(schema).toMatchSnapshot();
  });
});
