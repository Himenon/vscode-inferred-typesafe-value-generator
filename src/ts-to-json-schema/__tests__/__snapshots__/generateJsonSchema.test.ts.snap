// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSON Schema sample.ts 1`] = `
Object {
  "jsonSchema": Object {
    "$ref": "#/definitions/ServerSideConfig",
    "$schema": "http://json-schema.org/draft-07/schema#",
    "definitions": Object {
      "ServerSideConfig": Object {
        "additionalProperties": false,
        "properties": Object {
          "name": Object {
            "type": "string",
          },
          "service_a": Object {
            "$ref": "#/definitions/ServiceA",
          },
          "service_b": Object {
            "additionalProperties": false,
            "properties": Object {
              "apiToken": Object {
                "type": "string",
              },
              "timeout": Object {
                "type": "number",
              },
              "url": Object {
                "enum": Array [
                  "https://dev.example.com",
                  "https://prod.example.com/b",
                ],
                "type": "string",
              },
            },
            "required": Array [
              "url",
              "timeout",
            ],
            "type": "object",
          },
        },
        "required": Array [
          "name",
          "service_b",
        ],
        "type": "object",
      },
      "ServiceA": Object {
        "additionalProperties": false,
        "properties": Object {
          "url": Object {
            "type": "string",
          },
        },
        "required": Array [
          "url",
        ],
        "type": "object",
      },
    },
  },
}
`;

exports[`JSON Schema types.ts 1`] = `
Object {
  "jsonSchema": Object {
    "$ref": "#/definitions/ServerSideConfig",
    "$schema": "http://json-schema.org/draft-07/schema#",
    "definitions": Object {
      "ServerSideConfig": Object {
        "additionalProperties": false,
        "properties": Object {
          "name": Object {
            "type": "string",
          },
          "service_a": Object {
            "$ref": "#/definitions/ServiceA",
          },
          "service_b": Object {
            "additionalProperties": false,
            "properties": Object {
              "apiToken": Object {
                "type": "string",
              },
              "timeout": Object {
                "type": "number",
              },
              "url": Object {
                "enum": Array [
                  "https://dev.example.com",
                  "https://prod.example.com/b",
                ],
                "type": "string",
              },
            },
            "required": Array [
              "url",
              "timeout",
            ],
            "type": "object",
          },
        },
        "required": Array [
          "name",
          "service_b",
        ],
        "type": "object",
      },
      "ServiceA": Object {
        "additionalProperties": false,
        "properties": Object {
          "url": Object {
            "type": "string",
          },
        },
        "required": Array [
          "url",
        ],
        "type": "object",
      },
    },
  },
}
`;
